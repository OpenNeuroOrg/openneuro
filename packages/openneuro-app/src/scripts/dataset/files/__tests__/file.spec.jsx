import React from "react"
import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { LocalStorageProvider } from "../../../utils/local-storage.tsx"
import File from "../file"

const wrapper = ({ children }) => (
  <MemoryRouter>
    <LocalStorageProvider defaultValue={{ agreement: true }}>
      {children}
    </LocalStorageProvider>
  </MemoryRouter>
)

describe("File component", () => {
  it("renders with common props", () => {
    const { asFragment } = render(
      <File datasetId="ds001" path="" filename="README" size={500} />,
      { wrapper },
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("renders for dataset snapshots", () => {
    const { asFragment } = render(
      <File
        datasetId="ds001"
        snapshotTag="1.0.0"
        path=""
        filename="README"
        size={500}
      />,
      { wrapper },
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("generates correct download links for top level files", () => {
    render(<File datasetId="ds001" path="" filename="README" size={500} />, {
      wrapper,
    })
    expect(screen.getByRole("link", { name: "download file" })).toHaveAttribute(
      "href",
      "/crn/datasets/ds001/files/README",
    )
  })
  it("generates correct download links for nested files", () => {
    render(
      <File
        datasetId="ds001"
        path="sub-01:anat"
        filename="sub-01_T1w.nii.gz"
        size={2000000}
      />,
      { wrapper },
    )
    expect(screen.getByRole("link", { name: "download file" })).toHaveAttribute(
      "href",
      "/crn/datasets/ds001/files/sub-01:anat:sub-01_T1w.nii.gz",
    )
  })
})
