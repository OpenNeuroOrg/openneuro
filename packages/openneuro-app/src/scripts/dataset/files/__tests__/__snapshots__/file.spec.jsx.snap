// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`File component > renders for dataset snapshots 1`] = `
<DocumentFragment>
  <span
    aria-label=""
    class=" on-icon  "
    role="img"
    style="color: deepskyblue;"
  >
    <i
      class="fab fa-markdown"
    />
  </span>
    README
  <span
    class="filetree-editfile"
  >
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-medium "
    >
      <span
        class=" "
        data-flow="up"
        data-tooltip="Download: 500B"
      >
        <span
          class="edit-file download-file"
        >
          <a
            aria-label="download file"
            download="README"
            href="/crn/datasets/ds001/snapshots/1.0.0/files/README"
          >
            <i
              class="fa fa-download"
            />
          </a>
        </span>
      </span>
    </div>
    <span
      class=" "
      data-flow="up"
      data-tooltip="View"
    >
      <span
        class="edit-file view-file"
      >
        <a
          aria-label="view file"
          href="/datasets/ds001/versions/1.0.0/file-display/README"
        >
          <i
            class="fa fa-eye"
          />
        </a>
      </span>
    </span>
  </span>
</DocumentFragment>
`;

exports[`File component > renders with common props 1`] = `
<DocumentFragment>
  <span
    aria-label=""
    class=" on-icon  "
    role="img"
    style="color: deepskyblue;"
  >
    <i
      class="fab fa-markdown"
    />
  </span>
    README
  <span
    class="filetree-editfile"
  >
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-medium "
    >
      <span
        class=" "
        data-flow="up"
        data-tooltip="Download: 500B"
      >
        <span
          class="edit-file download-file"
        >
          <a
            aria-label="download file"
            download="README"
            href="/crn/datasets/ds001/files/README"
          >
            <i
              class="fa fa-download"
            />
          </a>
        </span>
      </span>
    </div>
    <span
      class=" "
      data-flow="up"
      data-tooltip="View"
    >
      <span
        class="edit-file view-file"
      >
        <a
          aria-label="view file"
          href="/datasets/ds001/file-display/README"
        >
          <i
            class="fa fa-eye"
          />
        </a>
      </span>
    </span>
  </span>
</DocumentFragment>
`;
